syntax = "proto2";
package tdrec.protos;

import "tdrec/protos/optimizer.proto";
import "tdrec/protos/model.proto";
import "tdrec/protos/dataset.proto";
import "tdrec/protos/feature.proto";

message PipelineConfig {
    required string model_dir = 1;

    required TrainConfig train_config = 2;

    required DatasetConfig dataset_config = 3;

    required FeatureConfig feature_config = 4;

    required ModelConfig model_config = 5;
}

message TrainConfig {
    // optimizer config
    required OptimizerConfig optimizer_config = 1;
    // number of epochs to train models
    optional uint32 num_epochs = 2 [default = 1];
    // the frequency the loss and lr will be logged during training
    optional uint32 log_step_count_steps = 3 [default = 1000];
    // Max checkpoints to keep
    optional uint32 keep_checkpoint_max = 4 [default = 10];
}
